python_modules/dagster/dagster/_core/definitions/data_version.py

```diff
@@ -508,3 +529,11 @@ def _is_volatile(self, *, key: AssetKey) -> bool:
         else:
             deps = self.asset_graph.get_parents(key)
             return len(deps) == 0 or any(self._is_volatile(key=dep_key) for dep_key in deps)
+
+    def _get_asset_record(self, *, key: AssetKey) -> Optional["AssetRecord"]:
```

Comment:
I think we can avoid this _get_asset_record method entirely (it's too specific, you just need the latest materialization event), and instead have something like:

```
def get_latest_materialization_or_observation_record(self, key):
    if not self.asset_graph.is_source(key) and not self.instance_queryer.has_cached_asset_record(key):
        ancestors = self.instance_queryer.prefetch_asset_records(ancestors)
        self.instance_queryer.prefetch_asset_records(ancestors)
    return self.instance_queryer.get_latest_materialization_or_observation_record(key)
```

this seems like the cleanest way to interface with the existing functionality imo

Reply:
Good suggestion, I've updated the PR to incorporate this.